google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ChecksumResultSet.java

```diff
@@ -354,6 +364,11 @@ private void funnelArray(
             into.putInt(stringValue.length());
             into.putUnencodedChars(stringValue);
             break;
+          case JSON:
+            Json json = (Json) value;
+            into.putInt(json.value.length());
```

Comment:
This will throw a `NullPointerException` if `json.value` is null (see also above)